src/config.ts

```diff
@@ -54,3 +54,46 @@ export async function resolveConnections(): Promise<ConnectionConfig[]> {
 
   return result;
 }
+
+/**
+ * Validate connection keys.
+ *
+ * @param {string[]} keys
+ * @returns {void}
+ */
+function validateConnections(keys: string[]): void {
+  const errors: string[] = [];
+  keys.forEach(element => {
+    if (!process.env[element]) {
+      errors.push(element);
+    }
+  });
+  if (errors.length) {
+    throw new Error(`Following environment variables were not set: ${errors}`);
+  }
+}
+
+/**
+ * Resolve database connections from Env.
+ *
+ * @param {string[]} keys
+ * @returns {ConnectionConfig[]}
+ */
+export function resolveConnectionsFromEnv(): ConnectionConfig[] {
+
+  validateConnections(ENV_KEYS);
+
+  const connection = {
```

Comment:
```suggestion
  const connection: ConnectionConfig = {
```

Reply:
Typescript is throwing error

 ```js
Type 'string | undefined' is not assignable to type 'string'.

Type 'undefined' is not assignable to type 'string'.ts(2322)
The expected type comes from property ${KEY} which is declared 
here on type 'ConnectionConfig'
```

```js    
const connection: ConnectionConfig = {}
```
 as `process.env[KEY]` might be `undefined` also. But the key have only `scalar` type in `ConnectionConfig` (No undefined type).

##### As we have already handled the `undefined` case in the above _`validateConnections`_ function.

Using  `{} as ConnectionConfig` shouldn't be a problem here.

```js
const connection = {} as ConnectionConfig;
```
@mesaugat @kabirbaidhya

Reply:
This is a reasonable typecheck so we shouldn't suppress this, but instead we should resolve the issue somewhere in the assignment.

Reply:
You're getting these issue because each of the keys with `process.env.*` can be either `string` or `undefined`. So, you'll need to null-coalesce here.